### ISSUE TITLE
Error checking if user exists - "TypeError: Cannot read property 'localId' of undefined"

### ISSUE THREAD

**Issue by aaronosher** [firebase/firebase-admin-node#1078]
### [READ] Step 1: Are you in the right place?

  * For issues related to __the code in this repository__ file a Github issue.
  * If the issue pertains to Cloud Firestore, read the instructions in the "Firestore issue"
    template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Operating System version: macOS 10.15.7
  * Firebase SDK version: 9.3.0
  * Firebase Product: auth
  * Node.js version: 12.19.0
  * NPM version: 6.14.8

### [REQUIRED] Step 3: Describe the problem

With the functions and authentication emulators, when running `auth().getUser()` with a UID that doesn't exist, I get `TypeError: Cannot read property 'localId' of undefined`. And not an error with code `auth/user-not-found` as per the documentation.

I have confirmed this is specific to the emulator, as I don't get the TypeError when I pushed my function to a production environment.

#### Steps to reproduce:

1. Initialise a firebase project with the authentication and functions emulator
2. Write a function that checks if a user exists by using `auth().getUser()`
3. Execute your function with the emulator

Edit: I've realised it's probably worthwhile adding the stack trace
```
TypeError: Cannot read property 'localId' of undefined
    at new UserRecord (functions/node_modules/firebase-admin/lib/auth/user-record.js:272:23)
    at functions/node_modules/firebase-admin/lib/auth/auth.js:133:20
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Object.exports.userExists (functions/lib/services/users.js:8:22)
    at async functions/lib/routes/discord.js:36:24
```

#### Relevant Code:

Here is the code I've used to check if a user exists

```TypeScript
import { auth } from 'firebase-admin';

export const userExists = async (uid: string): Promise<boolean> => {
  try {
    const user = await auth().getUser(uid);
    if (user) {
      return true;
    }

    return false;
  } catch (e) {
    if (e.code === 'auth/user-not-found') {
      return false;
    }
    throw e;
  }
};
```

**Comment by dsgriffin** [comment]
Also ran into this issue, reverting to `firebase-admin` version `9.2.0` resolves it temporarily.

**Comment by aaronosher** [comment]
I've temporarily updated my catch block to the following, with a note to revert it when the behaviour is patched.
```ts
catch (e) {
    if (e.code === 'auth/user-not-found') {
      return false;
    } else if (process.env.FUNCTIONS_EMULATOR) {
      return false;
    }
    throw e;
  }
```

**Comment by yuchenshi** [comment]
Hi all, this is an issue in the Auth Emulator and we're drafting a fix to be released in the CLI.

Also, friendly reminder that downgrading `firebase-admin` to 9.2.0 will NOT fix the issue -- 9.2.0 does not support Auth Emulator at all, and your requests WILL AFFECT PRODUCTION. Please wait and upgrade the CLI (not Admin SDK) once that is available.

**Comment by yuchenshi** [comment]
Fix coming in the next CLI release

**Comment by yuchenshi** [comment]
This should be fixed as of Firebase CLI v8.15.0. No update to `firebase-admin` needed.
